translator/GCP_cloud_function/cloud_function/main.py

```diff
@@ -42,28 +41,59 @@ def translate_newest_icone_to_wzdx(event, context):
   return
 
 def get_ftp_file(url) :
-  with closing(request.urlopen(url)) as r:
-    return r.read().decode('utf-8-sig')
-
+  return urllib.request.urlopen(url).read().decode('utf-8-sig')
 
 def parse_xml(xml_string):
   return xmltodict.parse(xml_string)
 
 def get_wzdx_schema(schema_file_name):
-  return json.loads(open(schema_file_name,'r').read())
+  try:
+    schema_string=open(schema_file_name,'r').read()
+  except FileNotFoundError as e:
+    raise RuntimeError('invalid schema: file does not exist')
+  if not schema_string:
+    raise RuntimeError('invalid schema: empty schema')
+  try:
+    schema_obj= json.loads(schema_string)
+    jsonschema.validate('', schema_obj)  
+  except json.decoder.JSONDecodeError as e :
+    raise RuntimeError('invalid schema: not valid json') from e
+  except jsonschema.SchemaError as e :
+    raise RecursionError('invalid schema: schema failed validation') from e
+  except jsonschema.ValidationError:
+    return schema_obj
+  return schema_obj
 
-#this function retrieves username and password info from secrets .This will be implemented after correct secrets are implemented.
 def get_ftp_credentials():
   secret_client = secretmanager.SecretManagerServiceClient()
   username_secret_name = os.environ['icone_ftp_username_secret_name']
   password_secret_name = os.environ['icone_ftp_password_secret_name']
   project_id = os.environ['project_id']
   request = {"name": f"projects/{project_id}/secrets/{username_secret_name}/versions/latest"}
-  response = client.access_secret_version(request)
+  response = secret_client.access_secret_version(request)
   username = response.payload.data.decode("UTF-8")
 
   request = {"name": f"projects/{project_id}/secrets/{password_secret_name}/versions/latest"}
-  response = client.access_secret_version(request)
+  response = secret_client.access_secret_version(request)
   password = response.payload.data.decode("UTF-8")
 
-  return username,password
\ No newline at end of file
+  return username,password
+
+
+def unsupported_messages_callback(message):
```

Comment:
Can we add some unit tests to show what happens with this method?

Reply:
i added another 2 test cases and pushed it to dev